### ISSUE TITLE
drone exec panic when insufficient privilege to connect to Docker

### ISSUE THREAD

**Issue by siddharthist** [drone/drone#1399]
When drone was denied permission to the docker socket, it exited with the following panic:

```
➤ drone exec

[DRONE] starting job #1
INFO[0000] Pulling image drone/drone-exec:latest        
ERRO[0000] Error pulling drone/drone-exec:latest. Post http://unix.sock/v1.15/images/create?fromImage=drone%2Fdrone-exec%3Alatest: dial unix /var/run/docker.sock: connect: permission denied

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x40674f]

goroutine 1 [running]:
main.run(0x7f8fbb5b4a88, 0xc82000ff20, 0xc82009e800, 0x6, 0x8, 0xc82008eb00, 0x53f, 0x0, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:232 +0x33f
main.execCmd(0xc820088fc0, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:182 +0x10fc
main.glob.func8(0xc820088fc0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:24 +0x28
github.com/drone/drone-cli/vendor/github.com/codegangsta/cli.Command.Run(0x8cf520, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x91ae50, 0x16, 0x0, ...)
        /drone/src/github.com/drone/drone-cli/vendor/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/drone/drone-cli/vendor/github.com/codegangsta/cli.(*App).Run(0xc820088d80, 0xc82000a1a0, 0x2, 0x2, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/vendor/github.com/codegangsta/cli/app.go:175 +0xfd6
main.main()
        /drone/src/github.com/drone/drone-cli/drone/main.go:44 +0x48f

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1
```

Unfortunately, `drone -v` just outputs `drone version`, but it is definitely the most recent release from `http://downloads.drone.io/drone-cli/drone_linux_amd64.tar.gz`.

**Comment by bradrydzewski** [comment]
> Post http://unix.sock/v1.15/images/create?fromImage=drone%2Fdrone-exec%3Alatest: dial unix /var/run/docker.sock: connect: permission denied

you need to make sure Docker is running on the default unix socket on your host machine at `/var/run/docker.sock`

**Comment by gtaylor** [comment]
I wonder if we should add an additional line or two to the error message suggesting some courses of actions. This one is probably going to be one of the more common setup mistakes.

**Comment by siddharthist** [comment]
@bradrydzewski It certainly is, and when this command is rerun with `sudo`, drone works as expected. I just figured that a panic with a nil pointer dereference wasn't the expected behavior when a permissions error is encountered.

**Comment by bradrydzewski** [comment]
@siddharthist can you `docker pull drone/drone-exec` to ensure you have the latest version of the build runner? The reason I believe the version is out of date is that the lines referenced in the stacktrace don't seem to correspond to latest code (one points to a comment, for example):
https://github.com/drone/drone-exec/blob/master/exec/exec.go#L182
https://github.com/drone/drone-exec/blob/master/exec/exec.go#L232

I still think we'll get a panic, but would be helpful if you could post an updated stacktrace which should help us pinpoint the exact location of the panic and expedite a fix.

**Comment by siddharthist** [comment]
@bradrydzewski Is the following what you meant? Also, allow me to recommend [panicwrap](https://github.com/mitchellh/panicwrap), which can help with logging panics, adding a message, etc.
```
sudo docker pull drone/drone-exec
[sudo] password for siddharthist: 
Using default tag: latest
latest: Pulling from drone/drone-exec
Digest: sha256:b554d9207a2575b746d1307cc76382cd05c478fc634cd9446037dc6d059adb57
Status: Image is up to date for drone/drone-exec:latest
distributive ➤ drone exec                       

[DRONE] starting job #1
INFO[0000] Pulling image drone/drone-exec:latest        
ERRO[0000] Error pulling drone/drone-exec:latest. Post http://unix.sock/v1.15/images/create?fromImage=drone%2Fdrone-exec%3Alatest: dial unix /var/run/docker.sock: connect: permission denied
 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x40674f]

goroutine 1 [running]:
main.run(0x7f091ff1ca88, 0xc82000ff20, 0xc82009e900, 0x6, 0x8, 0xc8200d6000, 0x610, 0x0, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:232 +0x33f
main.execCmd(0xc820088fc0, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:182 +0x10fc
main.glob.func8(0xc820088fc0)
        /drone/src/github.com/drone/drone-cli/drone/exec.go:24 +0x28
github.com/drone/drone-cli/vendor/github.com/codegangsta/cli.Command.Run(0x8cf520, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x91ae50, 0x16, 0x0, ...)
        /drone/src/github.com/drone/drone-cli/vendor/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/drone/drone-cli/vendor/github.com/codegangsta/cli.(*App).Run(0xc820088d80, 0xc82000a1a0, 0x2, 0x2, 0x0, 0x0)
        /drone/src/github.com/drone/drone-cli/vendor/github.com/codegangsta/cli/app.go:175 +0xfd6
main.main()
        /drone/src/github.com/drone/drone-cli/drone/main.go:44 +0x48f

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1
```

**Comment by bradrydzewski** [comment]
fixed in 0.5
